### ISSUE TITLE
Question on crowdstrike tagging kafkactl as highly vulnerable

### ISSUE THREAD

**Issue by govardha** [deviceinsight/kafkactl#144]
Hello,

I started using kafkactl and I LOVE it.  But when my infosec team "scanned" this in whatever systems they have(I am waiting for their reports).  One of the products they use is crowdstrike and it says this app is 100% danger.  

Has anybody really seen any sign of this?  I keep searching around and find nothing.  I am looking at the source code and I don't see anything crazy.  

I would greatly appreciate some insights from the developers.  I will post any details when I do get the scan reports eventually.

This one checksum file is getting flagged big time

https://github.com/deviceinsight/kafkactl/blob/main/go.sum

Thank you!
g

**Comment by d-rk** [comment]
Hey @govardha 
are you using the latest version (v3.0.2)?
In the latest version some of the dependencies have been updated to address this. See: https://github.com/deviceinsight/kafkactl/pull/143

**Comment by govardha** [comment]
Hello @d-rk 

Thanks for your response.  Here is the list of vulnerabilities that our internal scanners are pointing out.

I am running the latest version.

gov

critical:8.6    github.com/gogo/protobuf v1.2.1 Open    CVE-2021-3121   Direct
critical:8.6    github.com/gogo/protobuf v1.1.1 Open    CVE-2021-3121   Direct
critical:8.2    github.com/miekg/dns v1.0.14    Open    sonatype-2021-1401      Direct
critical:7.7    github.com/coreos/etcdv3.3.13+incompatible      Open    CVE-2020-15114  Direct
critical:7.7    github.com/coreos/etcdv3.3.10+incompatible      Open    CVE-2020-15114  Direct
critical:7.5    gopkg.in/yaml.v2v2.2.1  Open    CVE-2022-3064   Direct
critical:7.5    gopkg.in/yaml.v2v2.2.2  Open    CVE-2022-3064   Direct
critical:7.5    github.com/hashicorp/consul/sdk v0.1.1  Open    CVE-2022-29153  Direct
critical:7.5    golang.org/x/textv0.3.6 Open    CVE-2022-32149  Direct
critical:7.5    golang.org/x/textv0.3.2 Open    CVE-2021-38561  Direct
critical:7.5    golang.org/x/textv0.3.0 Open    CVE-2022-32149  Direct
critical:7.5    golang.org/x/textv0.3.2 Open    CVE-2020-14040  Direct
critical:7.5    github.com/coreos/etcdv3.3.13+incompatible      Open    CVE-2020-15115  Direct
critical:7.5    golang.org/x/textv0.3.6 Open    CVE-2021-38561  Direct
critical:7.5    golang.org/x/textv0.3.2 Open    CVE-2022-32149  Direct
critical:7.5    github.com/gorilla/websocket v1.4.0     Open    sonatype-2019-0666      Direct
critical:7.5    golang.org/x/textv0.3.3 Open    CVE-2021-38561  Direct
critical:7.5    golang.org/x/textv0.3.7 Open    CVE-2022-32149  Direct
critical:7.5    github.com/hashicorp/consul/api v1.1.0  Open    CVE-2022-29153  Direct
critical:7.5    github.com/coreos/etcdv3.3.10+incompatible      Open    CVE-2020-15115  Direct
critical:7.5    golang.org/x/textv0.3.0 Open    CVE-2020-14040  Direct
critical:7.5    golang.org/x/textv0.3.0 Open    CVE-2021-38561  Direct
critical:7.5    github.com/prometheus/client_golangv0.9.1       Open    CVE-2022-21698  Direct
critical:7.5    golang.org/x/textv0.3.3 Open    CVE-2022-32149  Direct
critical:7.5    github.com/prometheus/client_golangv0.9.3       Open    CVE-2022-21698  Direct
severe:6.5      gopkg.in/yaml.v2v2.2.2  Open    sonatype-2019-0900      Direct
severe:6.5      gopkg.in/yaml.v2v2.2.1  Open    sonatype-2019-0900      Direct
severe:6.5      github.com/coreos/etcdv3.3.13+incompatible      Open    CVE-2020-15136  Direct
severe:6.5      github.com/coreos/etcdv3.3.10+incompatible      Open    CVE-2020-15136  Direct
severe:6.5      github.com/pkg/sftpv1.10.1      Open    sonatype-2019-0890      Direct
severe:6.5      github.com/gorilla/sessions v1.2.1      Open    sonatype-2021-4899      Direct
severe:6.1      github.com/yuin/goldmark v1.1.32        Open    sonatype-2020-0584      Direct
severe:6.1      github.com/yuin/goldmark v1.1.27        Open    sonatype-2020-0584      Direct

**Comment by govardha** [comment]
So it definitely looks like there are a bunch of modules which have dependencies which have vulnerabilities.  I went through the list of modules and tried using a newer version of a bunch of modules and compiled the app and it compiles.  Unfortunately I did not test as I did this at home and I don't have access to a cluster here.

Here are the modules I upgraded to and what versions

```
github.com/golang/protobuf v1.4.2 -> v1.5.2
github.com/jhump/protoreflect  v1.10.1->v1.14.1
github.com/spf13/cobra v1.0.1-0.20200629195214-2c5a0d->v1.6.1
github.com/spf13/viper v1.7.1 -> v1.15.0
golang.org/x/sync v0.0.0-20220923202941-7f9b1623fab7 ->v0.1.0
google.golang.org/protobuf v1.25.1-0.20200805231151-a -> google.golang.org/protobuf v1.28.1
gopkg.in/yaml.v2 v2.3.0 -> v2.4.0

Indirect
github.com/cpuguy83/go-md2man/v2 v2.0.0 // indirect   |         github.com/cpuguy83/go-md2man/v2 v2.0.2 // indirect
github.com/fsnotify/fsnotify v1.4.9 // indirect       |         github.com/fsnotify/fsnotify v1.6.0 // indirect
github.com/google/go-cmp v0.5.5 // indirect           |
github.com/inconshreveable/mousetrap v1.0.0 // indire |         github.com/inconshreveable/mousetrap v1.0.1 // indire
github.com/magiconair/properties v1.8.4 // indirect   |         github.com/magiconair/properties v1.8.7 // indirect
github.com/mitchellh/mapstructure v1.3.3 // indirect  |         github.com/mitchellh/mapstructure v1.5.0 // indirect
github.com/pelletier/go-toml v1.8.1 // indirect       |         github.com/pelletier/go-toml/v2 v2.0.6 // indirect
github.com/russross/blackfriday/v2 v2.0.1 // indirect |         github.com/russross/blackfriday/v2 v2.1.0 // indirect
github.com/shurcooL/sanitized_anchor_name v1.0.0 // i |
github.com/spf13/afero v1.4.0 // indirect             |         github.com/spf13/afero v1.9.3 // indirecat
github.com/spf13/cast v1.3.1 // indirect              |         github.com/spf13/cast v1.5.0 // indirect
github.com/subosito/gotenv v1.2.0 // indirect         |         github.com/subosito/gotenv v1.4.2 // indirect
go.uber.org/atomic v1.5.1 // indirect                 |
golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83af |         google.golang.org/genproto v0.0.0-20221227171554-f968
google.golang.org/genproto v0.0.0-20200526211855-cb27 |         google.golang.org/genproto v0.0.0-20221227171554-f968
gopkg.in/ini.v1 v1.61.0 // indirect                   |         gopkg.in/ini.v1 v1.67.0 // indirect
                                                                gopkg.in/yaml.v3 v3.0.1 // indirect
```

**Comment by d-rk** [comment]
@govardha i just released v3.0.3 with updated dependencies
